Marlin/MarlinSerial.cpp

```diff
@@ -310,3 +315,86 @@ MarlinSerial customizedSerial;
 #if defined(USBCON) && ENABLED(BLUETOOTH)
   HardwareSerial bluetoothSerial;
 #endif
+
+#if ENABLED(EMERGENCY_PARSER)
+
+  // Currently looking for: M108, M112, M410
+  // If you alter the parser please don't forget to update the capabilities in Conditionals.h
+
+  void emergency_parser(unsigned char c) {
+
+    enum e_parser_state {
+      state_RESET,
+      state_N,
+      state_M,
+      state_M1,
+      state_M10,
+      state_M11,
+      state_M4,
+      state_M41,
+      state_IGNORE // to '\n'
+    };
+
+    static e_parser_state state = state_RESET;
+
+    if (c == '\n') state = state_IGNORE;
+
+    switch (state) {
+      case state_RESET:
+        switch (c) {
+          case ' ': break;
+          case 'N': state = state_N;      break;
+          case 'M': state = state_M;      break;
+          default: state = state_IGNORE;
+        }
+        break;
```

Comment:
Why only ignore `G` and `T`? What about a command (as artificial as this may be) like "`Hey there M108!`". Although it doesn't start with `G` or `T` it would still proceed to `state_M108`. We know `N` and ` ` are ok before 'M'. While _everything else_ can be ignored.